src/ui/layout.tsx

```diff
@@ -22,6 +22,71 @@ import { PathMatch } from "@solidjs/router/dist/types";
 
 const PROJECTS = ["solid-router", "solid-start", "solid-meta"] as const;
 
+export enum Project {
+	Core = "",
+	Router = "/solid-router",
+	Start = "/solid-start",
+	Meta = "/solid-meta",
+}
+
+interface CurrentRouteMetaData {
+	project: Project | null;
+	locale: string;
+	isProjectRoot: boolean;
+}
+
+export const useCurrentRouteMetaData = (): CurrentRouteMetaData => {
+	let currentPath = useLocation().pathname;
+
+	// Trim trailing slash
+	currentPath = currentPath.endsWith("/")
+		? currentPath.slice(0, -1)
+		: currentPath;
+
+	const pathParts = currentPath.split("/").filter(Boolean);
+	const projectOrLocale: string = pathParts[0];
+
+	let returnObject: CurrentRouteMetaData = {
+		isProjectRoot: true,
+		locale: "",
+		project: null,
+	};
+
+	if (SUPPORTED_LOCALES.includes(projectOrLocale)) {
+		if (pathParts.length > 2) {
+			returnObject.isProjectRoot = false;
+		}
+
+		returnObject.locale = projectOrLocale;
+		checkPathBeyondLocale(pathParts[1] ?? "");
+	} else {
+		if (pathParts.length > 1) {
+			returnObject.isProjectRoot = false;
+		}
+
+		checkPathBeyondLocale(pathParts[0] ?? "");
+	}
+
+	function isInProjectEnum(projectPath: string): boolean {
+		return Object.values(Project).includes(projectPath as Project);
+	}
+
+	function checkPathBeyondLocale(path: string) {
+		if (path.length > 0) {
+			path = "/" + path;
+		}
+
+		if (isInProjectEnum(path)) {
+			returnObject.project = path as Project;
+		} else {
+			returnObject.isProjectRoot = false;
+			returnObject.project = "" as Project;
+		}
+	}
+
+	return returnObject;
+};
+
```

Comment:
This is a small nitpick but would it make sense to move this to the helpers file? Considering it is also being imported there, it would probably be better to have it there or a separate file than increasing the size of this UI file 🤔
Let me know your thoughts please!

Aside from this comment everything else is working fine for me and this seems a great improvement 😄

Reply:
That is a great idea!  I'll move it right away.